### PR TITLE
Added a script to download k3s kubeconfig for preview-environments

### PR DESCRIPTION
**Pull request by wulfthimm**
## Description

This adds the script `install-k3s-kubeconfig.sh`. It connects to the k3s cluster inside the preview VM and downloads the kubeconfig `k3s.yaml` replacing the context name and endpoint. The script reuses a lot of parts from other scripts and we might refactor it to reduce code.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/644

## How to test

Start a new workspace and push a new commit that creates a VM.

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
@wulfthimm: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **akosyakov** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @akosyakov` in a comment when ready.

Associated issue: *#644*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ArthurSens** [comment]
> The script reuses a lot of parts from other scripts and we might refactor it to reduce code.

What do you think of doing that in this PR? 🙂 

Let's refactor early instead of letting it get out of control 😅

**Comment by wulfthimm** [comment]
> > The script reuses a lot of parts from other scripts and we might refactor it to reduce code.
> 
> What do you think of doing that in this PR? slightly_smiling_face
> 
> Let's refactor early instead of letting it get out of control sweat_smile

That is true but out of the scopre of the issue.

**Comment by ArthurSens** [comment]
I believe any method that is duplicated can live in an "utils" or "lib" shell, then you can use `source <script>.sh` to import those methods, right?

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
I agree that we'll soon want a lib folder with files containing common functions that we can `source`, but I don't think quite yet. There is a lot of duplication that can be removed tough and I have suggested ways of doing it in the review :)

File: dev/preview/install-k3s-kubeconfig.sh
Comment (wulfthimm) [reply dev/preview/install-k3s-kubeconfig.sh:54]: I would rather have self-consistent scripts but sure.

File: dev/preview/install-k3s-kubeconfig.sh
Comment (wulfthimm) [reply dev/preview/install-k3s-kubeconfig.sh:47]: In case that this does not work during boot or someone is working on an other system I would like to keep those checks.

Comment (mads-hartmann) [reply dev/preview/install-k3s-kubeconfig.sh:43]: This is access to the Harvester Kubernetes "cluster" so it doesn't care about individual namespaces or VMs that have been created in the cluster - but we can clean up the scripts later so we don't have this check so many places.

**Review by mads-hartmann — APPROVED** [review]
Works like a charm.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bc38b1d66b91c3ce41c9ef871d2dee7022626e94</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7893#pullrequestreview-868882800" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **akosyakov** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @akosyakov` in a comment when ready.

Associated issue: *#644*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: dev/preview/install-k3s-kubeconfig.sh
Comment (wulfthimm) [reply dev/preview/install-k3s-kubeconfig.sh:43]: You are right, of course. My reply was wrong.

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Review by mads-hartmann — APPROVED** [review]
Thanks 🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 63cd08f60734d23c1300beffe68acddcad60026f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7893#pullrequestreview-868952607" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **akosyakov** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @akosyakov` in a comment when ready.

Associated issue: *#644*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7893#pullrequestreview-868952607" title="Approved">mads-hartmann</a>*

Associated issue: *#644*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
